awx/api/views/__init__.py

```diff
@@ -2563,10 +2563,37 @@ def _validate_spec_data(new_spec, old_spec):
                         return Response(dict(error=_(
                             "The {min_or_max} limit in survey question {idx} expected to be integer."
                         ).format(min_or_max=key, **context)))
-            if qtype in ['multiplechoice', 'multiselect'] and 'choices' not in survey_item:
-                return Response(dict(error=_(
-                    "Survey question {idx} of type {survey_item[type]} must specify choices.".format(**context)
-                )))
+            # if it's a multiselect or multiple choice, it must have coices listed
+            # choices and defualts must come in as strings seperated by /n characters.
+            if qtype == 'multiselect' or qtype == 'multiplechoice':
+                if 'choices' in survey_item:
+                    survey_item['choices'] = '\n'.join(choice for choice in survey_item['choices'].splitlines() if choice.strip() != '')
+                elif 'choices' not in survey_item:
+                    return Response(dict(error=_(
+                        "Survey question {idx} of type {survey_item[type]} must specify choices.".format(**context)
+                    )))
+                # If there is a default, split it out removing extra /n characters.
+                # This should happen for both multiple choice and multiselect, even though multiple choice hsould only have one default.
+                # There can still be extra newline characters added in the API.
+                if 'default' in survey_item:
+                    survey_item['default'] = '\n'.join(choice for choice in survey_item['default'].splitlines() if choice.strip() != '')
+                    list_of_defaults = survey_item['default'].splitlines()
+                    if qtype == 'multiplechoice':
+                        if len(list_of_defaults) > 1:
+                            return Response(dict(error=_(
+                                "Multiple Choice (Single Select) can only have one default value.".format(**context)
+                            )))
+                        if survey_item['default'] not in survey_item['choices']:
+                            return Response(dict(error=_(
+                                "Default choices must be answers from the choices listed.".format(**context)
+                            )))
+                    if qtype == 'multiselect':
+                        # If there is one or more default values in a multiselect loop through comparing each with the available choices.
+                        for item in list_of_defaults:
+                            if item not in survey_item['choices']:
+                                return Response(dict(error=_(
+                                    "Default choices must be answers from the choices listed.".format(**context)
+                                )))
```

Comment:
There is some consolidation that can happen in these clauses because they take place in the larger conditional block started on L2568, checking that the `qtype` is multiplechoice or multiselect.  I think that you could do away with the check on L2590 entirely, and then reduce the check that the default(s) are in the choices to a single loop.

A nice way to make things further compact would be to use Python's `any()` function: `if any(item not in survey_item['choices'] for item in list_of_defaults):`.

Reply:
I agree

> Default choices must be answers from the choices listed

just becomes one check, right after `list_of_defaults` is defined